### PR TITLE
Adding all relevant files for the Funky40

### PR DESCRIPTION
**Pull request by TheFourthCow**
This should add all proper files for the funky40 a keyboard I designed for myself, /u/TheFourthcow, a 40% ortholinear with split spacebar utilizing a pro micro and should also work with an elite c.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR -N/A - initial addition of new keyboard

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/funky40/config.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
```
Comment (zvecr) [inline keyboards/funky40/config.h:55]:
```suggestion
// #define IS_COMMAND() (get_mods() == MOD_MASK_SHIFT)
```

File: keyboards/funky40/config.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+/* key matrix pins */
+#define MATRIX_ROW_PINS { D4, C6, B4, B5 }
+#define MATRIX_COL_PINS { D3, F5, F4, F7, B1, B6, B2, B3, D2, F6, E6, D7 }
+#define UNUSED_PINS 
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (zvecr) [inline keyboards/funky40/config.h:44]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/funky40/config.h [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef CONFIG_H
+#define CONFIG_H
```
Comment (zvecr) [inline keyboards/funky40/config.h:18]:
```suggestion
#pragma once
```

File: keyboards/funky40/config.h [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
+
+#endif
```
Comment (zvecr) [inline keyboards/funky40/config.h:60]:
```suggestion

```

File: keyboards/funky40/funky40.h [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#ifndef KB_H
+#define KB_H
```
Comment (zvecr) [inline keyboards/funky40/funky40.h:2]:
```suggestion
#pragma once
```

File: keyboards/funky40/funky40.h [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+#include "quantum.h"
+
+#define KEYMAP( \
```
Comment (zvecr) [inline keyboards/funky40/funky40.h:6]:
```suggestion
#define LAYOUT( \
```

File: keyboards/funky40/funky40.h [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+	K200, K201, K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, \
+	K300, K301, K302, K303, K304, K305, K306, K307, K308, K309  \
+) { \
+	{ K000,  K001,  K002,  K003,  K004,  K005,  K006,  K007,  K008,  K009,  K010,  K011 }, \
+	{ K100,  K101,  K102,  K103,  K104,  K105,  K106,  K107,  K108,  K109,  K110,  K111 }, \
+	{ K200,  K201,  K202,  K203,  K204,  K205,  K206,  K207,  K208,  K209,  K210,  K211 }, \
+	{ K300,  K301,  K302,  K303,  K304,  K305,  K306,  K307,  K308,  K309,  KC_NO, KC_NO }  \
+}
+
+#endif
```
Comment (zvecr) [inline keyboards/funky40/funky40.h:18]:
```suggestion
```

File: keyboards/funky40/readme [hunk 8]
```diff
@@ -0,0 +1,14 @@
+# Funky40
```
Comment (zvecr) [inline keyboards/funky40/readme:1]: This file needs to be renamed `readme.md`

File: keyboards/funky40/readme [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+![Funky40](http://i.imgur.com/q2M3uEU.jpg) -UPDATE THIS
+
+A compact 40% (12x4) ortholinear keyboard kit made and sold by /u/TheFourthCow
+
+* Keyboard Maintainer: [/u/TheFourthCow](https://github.com/TheFourthCow)
+* Hardware Supported: Funky40 with Pro Micro or Elite C
+* Hardware Availability: TBD - Currently from /u/TheFourthCow pm me on reddit!
+Make example for this keyboard (after setting up your build environment):
+
+    make funky40:v1.1
```
Comment (zvecr) [inline keyboards/funky40/readme:12]:
```suggestion
    make funky40:default
```

File: keyboards/funky40/rules.mk [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+BOOTMAGIC_ENABLE ?= yes	# Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE ?= yes	# Mouse keys(+4700)
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
+COMMAND_ENABLE ?= no    # Commands for debug and configuration - Also allows space cadet shift enter to work - when held rshift when tapped enter
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= no
```
Comment (zvecr) [inline keyboards/funky40/rules.mk:26]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```

File: keyboards/funky40/keymaps/v1.1/keymap.c [hunk 11]
```diff
@@ -0,0 +1,39 @@
+/* Copyright 2020 /u/TheFourthCow
```
Comment (zvecr) [inline keyboards/funky40/keymaps/v1.1/keymap.c:1]: Keyboards **must** have a `default` keymap, which my suggestion is to rename the `v1.1` folder to `default`

File: keyboards/funky40/keymaps/v1.1/keymap.c [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "funky40.h"
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	KEYMAP(
```
Comment (zvecr) [inline keyboards/funky40/keymaps/v1.1/keymap.c:21]:
```suggestion
	LAYOUT(
```

File: keyboards/funky40/keymaps/v1.1/keymap.c [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+		KC_ESC, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_BSPC, 
+		KC_TAB, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, 
+		KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_DOT, KC_DOT, KC_SLSH, KC_SFTENT, //kc_sftent is a Space Cadet mod tap code that should allow right shift to be enter when tapped right shift when held
+		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, LT(1, KC_DEL), KC_SPC, KC_LEFT, KC_UP, KC_DOWN, KC_RGHT),
+										   //^This lt keycode is supposed to make this key delete when tapped and enable a momentary layer to access the next layer down when held
+	KEYMAP(
```
Comment (zvecr) [inline keyboards/funky40/keymaps/v1.1/keymap.c:27]:
```suggestion
	LAYOUT(
```

File: keyboards/funky40/keymaps/v1.1/keymap.c [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "funky40.h"
```
Comment (zvecr) [inline keyboards/funky40/keymaps/v1.1/keymap.c:17]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/funky40/keymaps/v1.1/readme [hunk 15]
```diff
@@ -0,0 +1,5 @@
+![Funky40 Layout Image](http://i.imgur.com/7Capi8W.png) -UPDATE THIS
```
Comment (zvecr) [inline keyboards/funky40/keymaps/v1.1/readme:1]: Rename to `readme.md`

**Comment by fauxpark** [comment]
Closing in favour of #8942

**Pull request closed without merging**